JavaFX Groovy classpath fix

acisternino opened the pull request:
This PR fixes a problem when compiling Groovy programs using the JavaFX technology from Oracle.
When using the Oracle JVM, Groovy code was being compiled without adding the Extension Classloader to the compile classpath.
The problem is described in [a post](https://groups.google.com/forum/#!topic/gradle-dev/q3u-T09km2s) on the gradle-dev Google group.

Integration tests are included.

bigdaz commented:
This PR has been merged. We always intended for the ext classLoader to be included when compiling Groovy: this was already done for Java compilation.

 I changed the implementation to use `ClassLoader.systemClassLoader.parent` to determine the Extension class loader, and added a test for compiling Java sources that use JavaFX (on Java8).

Thanks for the contribution. This will be in Gradle 2.4

acisternino commented:
Hi,

Great news, thanks a lot!

On 4 March 2015 at 21:47, Daz DeBoer <notifications@github.com> wrote:

> This PR has been merged. We always intended for the ext classLoader to be
> included when compiling Groovy: this was already done for Java compilation.
>
> I changed the implementation to use ClassLoader.systemClassLoader.parent
> to determine the Extension class loader, and added a test for compiling
> Java sources that use JavaFX (on Java8).
>
> Thanks for the contribution. This will be in Gradle 2.4
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/gradle/gradle/pull/332#issuecomment-77243763>.
>



-- 
Andrea Cisternino, Erlangen, Germany
LinkedIn: http://www.linkedin.com/in/andreacisternino
GitHub: http://github.com/acisternino
